packages/web/src/lib/codemirrorLanguage.ts

```diff
@@ -0,0 +1,230 @@
+import { shader } from "@codemirror/legacy-modes/mode/clike";
+import { shell } from "@codemirror/legacy-modes/mode/shell";
+import { squirrel } from "@codemirror/legacy-modes/mode/clike";
+import { swift } from "@codemirror/legacy-modes/mode/swift";
+import { tcl } from "@codemirror/legacy-modes/mode/tcl";
+import { textile } from "@codemirror/legacy-modes/mode/textile";
+import { stex } from "@codemirror/legacy-modes/mode/stex";
+import { toml } from "@codemirror/legacy-modes/mode/toml";
+import { turtle } from "@codemirror/legacy-modes/mode/turtle";
+// import { twig } from "@codemirror/legacy-modes/mode/twig"; // not present anymore
+import { vb } from "@codemirror/legacy-modes/mode/vb";
+import { vbScript } from "@codemirror/legacy-modes/mode/vbscript";
+import { velocity } from "@codemirror/legacy-modes/mode/velocity";
+import { verilog } from "@codemirror/legacy-modes/mode/verilog";
+import { vhdl } from "@codemirror/legacy-modes/mode/vhdl";
+// import { wast } from "@codemirror/legacy-modes/mode/wast"; // available in CodeMirror 6
+import { webIDL } from "@codemirror/legacy-modes/mode/webidl";
+import { xQuery } from "@codemirror/legacy-modes/mode/xquery";
+import { languageMetadataMap } from "@/lib/languageMetadata";
+
+
+export type CodeMirrorLanguageKey = keyof typeof codemirrorLanguageMap;
+
+export const getCodemirrorLanguage = (linguistLanguage: string): StreamLanguage<unknown> | LanguageSupport | null => {
+    if (languageMetadataMap[linguistLanguage]?.codemirrorLanguage) {
+        const codemirrorLanguage = languageMetadataMap[linguistLanguage].codemirrorLanguage;
+        if (codemirrorLanguageMap[codemirrorLanguage]) {
+            return codemirrorLanguageMap[codemirrorLanguage];
+        }
+    }
+    return null;
+}
+
+export const codemirrorLanguageMap = {
+    // CodeMirror 6 languages
+    "brainfuck": brainfuck(),
+    "css": css(),
+    "c": cpp(),
+    "c#": csharp(),
+    "dot": dot(),
+    "elixir": elixir(),
+    "go": go(),
+    "graphql": graphqlLanguageSupport(),
+    "handlebars": new LanguageSupport(handlebarsLanguage),
+    "hcl": hcl(),
+    "html": html(),
+    "j": j(),
+    "java": java(),
+    "jsx": javascript({ jsx: true, typescript: false }),
+    "jq": jq(),
+    "json": json(),
+    "jsonc": jsonc(),
+    "less": less(),
+    "liquid": liquid(),
+    "makefile": makefile(),
+    "markdown": markdown(),
+    "mermaid": mermaid(), // not recognized by linguist/enry
+    "nix": nix(),
+    //  "perl": perl(), // not working - use codemirror 5 mode
+    "php": php(),
+    "prolog": prolog(),
+    "python": python(),
+    //  "r": r(), // not working - use codemirror 5 mode
+    "rust": rust(),
+    "sass": sass(),
+    "solidity": solidity,
+    "sparql": sparql(),
+    "spreadsheet": spreadsheet(),
+    "sql": sql(),
+    "svelte": svelte(),
+    "tsx": javascript({ jsx: true, typescript: true }),
+    "twig": twig(),
+    "typescript": javascript({ jsx: false, typescript: true }),
```

Comment:
nit: do we need a `javascript` key that defines `javascript({ jsx: false, typescript: false })`?

Reply:
ah I see what you mean - I see that `JavaScript` is being mapped to `jsx` so no worries, you can disregard my comment.